#!/usr/bin/env python
PACKAGE = "topico"

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("V_max",                       double_t, 0, "Maximum Velocity (m/s)",                 1.00,   0.00,  10.00)
gen.add("V_min",                       double_t, 0, "Minimum Velocity (m/s)",                -1.00, -10.00,   0.00)
gen.add("A_max",                       double_t, 0, "Maximum Acceleration (m^2/s)",           0.50,   0.00,   5.00)
gen.add("A_min",                       double_t, 0, "Minimum Acceleration (m^2/s)",          -0.50,  -5.00,   0.00)
gen.add("J_max",                       double_t, 0, "Maximum Jerk (m^3/s)",                   1.00,   0.00,  10.00)
gen.add("J_min",                       double_t, 0, "Minimum Jerk (m^3/s)",                  -1.00, -10.00,   0.00)

gen.add("b_sync_V",                    bool_t,   0, "sync trajectories with V?",                             True )
gen.add("b_sync_A",                    bool_t,   0, "sync trajectories with A?",                             True )
gen.add("b_sync_J",                    bool_t,   0, "sync trajectories with J?",                             False)
gen.add("b_sync_W",                    bool_t,   0, "sync trajectories with W?",                             False)

gen.add("b_rotate",                    bool_t,   0, "Rotate trajectory?",                                    False)
gen.add("b_hard_V_lim",                bool_t,   0, "Hard velocity limits?",                                 False)
gen.add("b_catch_up",                  bool_t,   0, "Catch up trajectories?",                                False)

gen.add("ts_rollout",                  double_t, 0, "Timesteps of the trajectory rollout",    0.10,   0.01,  10.00)
gen.add("direction",                   int_t,    0, "Direction of undefined DoF",             0   ,  -1   ,  1    )


exit(gen.generate(PACKAGE, "topico", "TopiCo"))
